src/coreclr/inc/gc_unwind_x86.h

```diff
@@ -47,6 +43,293 @@ enum RegMask
+    void doBigAndAssign(const ptrArgTP&);
+    void doBigOrAssign(const ptrArgTP& arg);
+    BOOL doBigEquals(const ptrArgTP&) const;
+    BOOL doBigIntersect(const ptrArgTP&) const;
+
+public:
+    ptrArgTP()
+    {
+        LIMITED_METHOD_CONTRACT;
+        SUPPORTS_DAC;
+
+        m_val = 0;
+    }
+
+    explicit ptrArgTP(ChunkType arg)
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        if (arg > MaxVal)
+        {
+            doBigInit(arg);
+        }
+        else
+        {
+            m_val = ChunkType(arg << 1);
+        }
+    }
+
+    ptrArgTP(ChunkType arg, UINT shift)
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        if ((arg > MaxVal) || (shift >= SMALL_BITS) || (arg > (MaxVal >> shift)))
+        {
+            doBigInit(arg);
+            doBigLeftShiftAssign(shift);
+        }
+        else
+        {
+            m_val = ChunkType(arg << (shift+1));
+        }
+    }
+
+    ptrArgTP operator &(const ptrArgTP& arg) const
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        ptrArgTP ret = *this;
+        ret &= arg;
+        return ret;
+    }
+
+    BOOL operator ==(const ptrArgTP& arg) const
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        if ((m_val | arg.m_val) & IS_BIG)
+        {
+            return doBigEquals(arg);
+        }
+        else
+        {
+            return m_val == arg.m_val;
+        }
+    }
+
+    BOOL operator !=(const ptrArgTP& arg) const
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        return !(*this == arg);
+    }
+
+    void operator <<=(unsigned shift)
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        if ((m_val == 0) || (shift == 0))     // Zero is a special case, don't need to do anything
+            return;
+
+        if (isBig() || (shift >= SMALL_BITS) || (m_val > (MaxVal >> (shift-1))))
+        {
+            doBigLeftShiftAssign(shift);
+        }
+        else
+        {
+            m_val <<= shift;
+        }
+    }
+
+    void operator >>=(unsigned shift)
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        if (isBig())
+        {
+            doBigRightShiftAssign(shift);
+        }
+        else
+        {
+            m_val >>= shift;
+            m_val &= ~IS_BIG;  // clear the isBig bit if it got set
+        }
+    }
+
+    void operator |=(const ptrArgTP& arg)
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        if (((m_val | arg.m_val) & IS_BIG) != 0)
+        {
+            doBigOrAssign(arg);
+        }
+        else
+        {
+            m_val |= arg.m_val;
+        }
+    }
+
+    void operator &=(const ptrArgTP& arg)
+    {
+        WRAPPER_NO_CONTRACT;
+        SUPPORTS_DAC;
+
+        if (((m_val | arg.m_val) & IS_BIG) != 0)
+        {
+            doBigAndAssign(arg);
+        }
+        else
+        {
+            m_val &= arg.m_val;
+        }
+    }
+
+    friend BOOL isZero(const ptrArgTP& arg)
```

Comment:
What are these `friend`s for?

Reply:
I took it from the original bitvector. These methods are not used as member methods. I can change it but I wanted to limit the number of changes.

Reply:
I meant what does `friend` modifier on a method implementation do in C++? I am surprised that it compiles, but I cannot think of it can do.

I would not mind deleting unnecessary cruft. I agree that it is best to avoid any logic changes.

Reply:
I *think* it is basically equivalent to declaring `friend BOOL isZero(const ptrArgTP& arg);` here and then placing the method itself outside of the `struct`/`class`.